docs/topics/fun-interfaces.md

```diff
@@ -63,6 +63,45 @@ fun main() {
 
 You can also use [SAM conversions for Java interfaces](java-interop.md#sam-conversions).
 
+## Migration from interface with constructor function to functional interface
+
+> Support for callable references to functional interface constructors is [Experimental](components-stability.md).
+> It may be dropped or changed at any time. Opt-in is required (see details below), and you should use it only for evaluation purposes.
+> We would appreciate your feedback on it in [YouTrack](https://youtrack.jetbrains.com/issue/KT-47939).
+>
+{type="warning"}
+
+Starting from 1.6.20, Kotlin supports callable references to functional interface constructors.
+This support adds a source-compatible way to migrate from an interface with a constructor function to a functional interface.
+Consider this "legacy" code:
+
+```kotlin
+interface KRunnable { 
+    fun invoke() 
+}
+
+fun KRunnable(block: () -> Unit): KRunnable = object : KRunnable { override fun invoke() = block() }
+```
+
+To migrate this code to the functional interface `KRunnable`, use the following code:
+
+```kotlin
+fun interface KRunnable { 
+    fun invoke()
+}
+```
+
+With callable references to functional interface constructors enabled, any code using the `::KRunnable` function reference will compile.
+Preserve the binary compatibility by marking the legacy function `KRunnable` with the [`@Deprecated`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-deprecated/)
+annotation with `DeprecationLevel.HIDDEN`:
+
+```kotlin
+@Deprecated(message = "Your message about the depreciation", level = DeprecationLevel.HIDDEN)
```

Comment:
`deprecation` and `depreciation` are two different words.

Reply:
Just a typo, thanks.